File: frontend/src/services/studyService.ts
import axios from 'axios';
import { StudyDeck } from '../types/models';

const API_URL = '/api/v1/study';

/**
 * Ruft alle Karten eines Decks in zufälliger Reihenfolge ab.
 * 
 * @param deckId Die ID des Decks
 * @returns Ein StudyDeck mit den Karten in zufälliger Reihenfolge
 */
export const getRandomizedDeck = async (deckId: string): Promise<StudyDeck> => {
  const response = await axios.get(`${API_URL}/decks/${deckId}/cards`);
  return response.data;
};

File: src/main/java/com/fao/flashcards/domain/service/StudyService.java
package com.fao.flashcards.domain.service;

import com.fao.flashcards.api.dto.StudyDeckDTO;
import java.util.Optional;

/**
 * Service-Interface für den Lernmodus.
 * Definiert die Methoden, die für den Lernmodus benötigt werden.
 */
public interface StudyService {
    
    /**
     * Gibt alle Karten eines Decks in zufälliger Reihenfolge zurück.
     * 
     * @param deckId Die ID des Decks
     * @return Ein Optional mit StudyDeckDTO mit den Karten in zufälliger Reihenfolge
     *         oder Optional.empty(), wenn das Deck nicht gefunden wurde
     */
    Optional<StudyDeckDTO> getRandomizedDeck(String deckId);
    
    /**
     * Prüft, ob ein Deck existiert.
     * 
     * @param deckId Die ID des Decks
     * @return true, wenn das Deck existiert, sonst false
     */
    boolean doesDeckExist(String deckId);
}